src/main/java/io/airlift/slice/Slices.java

```diff
@@ -67,6 +67,9 @@ public static Slice ensureSize(Slice existingSlice, int minWritableBytes)
             }
             else {
                 newCapacity *= SLICE_ALLOW_SKEW;
+                if (newCapacity > MAX_ARRAY_SIZE && minWritableBytes <= MAX_ARRAY_SIZE) {
```

Comment:
Unrelated to this change, but the line that says `newCapacity *= SLICE_ALLOW_SKEW` is not at all obvious. It has a couple of issues:
* It's multiplying an int with a double and assigning back to an int, which can produce unexpected results due to truncation/saturation
* It works because a cast from double -> int is saturating.

We should add a comment describing that or make the code more explicit.

Reply:
I agree, to me it wasn't obvious at first. 

> which can produce unexpected results due to truncation/saturation

This doesn't has much impact besides tests, because `SLICE_ALLOW_SKEW` is an arbitrary multiplier. Rounding of intermediate values are harmless.

> It works because a cast from double -> int is saturating

Would it be enough if I add a comment: `// double to int cast is saturating` to that line?